### ISSUE TITLE
Could not create transaction. Unexpected empty response from Firefly III. Check the logs.

### ISSUE THREAD

**Issue by Samdotr** [firefly-iii/firefly-iii#5993]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [ ] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Hello,

I've just got Firefly III 5.7.1 and FIDI running in a Docker container. I am currently trying to import data using a CSV (crafting my own, but for this example I've kept it simple with only one entry).

I've got the CSV to a stage where it is processed without errors, but cannot be imported due to the above error "Could not create transaction. Unexpected empty response from Firefly III. Check the logs.".

I've seen this error mentioned in [issue 3952](https://github.com/firefly-iii/firefly-iii/issues/3952) and [issue 4222](https://github.com/firefly-iii/firefly-iii/issues/4222) but neither has helped resolve my issue - I don't use a subdirectory or redirect with either the Firefly III instance or FIDI. The OP of issue 4222 said they'd fixed it by changing environment variables but didn't mention which they changed.

### Expected behaviour

I haven't had a successful import yet but would expect it to import without errors 😄 

### Debug information

Debug information generated at 2022-04-12 11:17:35 Europe/London for Firefly III version **5.7.1**.

| Scope | Version |
| --- | --- |
| Firefly III | 5.7.1 |
| Firefly III API | 1.5.6 |
| Build | 153, 04-04-2022 05:56:25 CEST |
| DB version | 18 (exp. 18) |
| PHP | `8.0.17` |
| Host | `Linux` |

| System info | Value |
| --- | --- |
| System TZ | Europe/London |
| Browser TZ | Europe/London |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | notice, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| Default language | en_GB |
| Default locale | equal |
| BCscale | 24 |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | mysql |
| Trusted proxies (.env) | `**` |

| User info | Value |
| --- | --- |
| Login provider | eloquent  |
| User guard | web  |
| Headers | N/A, N/A  |
| Stateful domains |  |
| Session start | 2022-04-01 00:00:00 |
| Session end | 2022-04-30 23:59:59 |
| Session first | 2022-04-12 00:00:00 |
| User ID | 1 |
| User language | en_GB |
| User locale | en_GB |
| Attempt at "en_GB.utf8" | 'en_GB.utf8' |
| Attempt at "en_GB.UTF-8" | 'en_GB.UTF-8' |
| User agent | Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.79 Safari/537.36 |

### Steps to reproduce

1. Upload CSV and configuration file to FIDI page and click Next
2. Click submit on configuration page
3. Click submit on role definition page
4. Click submit on map data page
5. Click start job
6. "The conversion routine has finished 🎉. Please wait to be redirected!" click start job again
7. "The import routine has finished 🎉. Done!

Error(s)
Line #0: Could not create transaction. Unexpected empty response from Firefly III. Check the logs."

### Additional info

For reference I've attached log files, the original CSV, the import configuration JSON file and the converted output:

Firefly app Docker log: https://pastebin.com/DHg5Az1J
FIDI Docker log: https://pastebin.com/wQERbCNa
Original (test) CSV: https://pastebin.com/RxJfn5RY
Import configuration JSON file: https://pastebin.com/4p75UFqc
Converted JSON output: https://pastebin.com/EwYXDBQm

**Comment by JC5** [comment]
See also https://github.com/firefly-iii/firefly-iii/discussions/5986

A solution is on the way.

**Comment by Samdotr** [comment]
When was the data importer last updated? I deployed my setup for the first time last night (2022-04-11 21:19:05 GMT+1) so would presume it is up to date unless an update was pushed today

**Comment by Samdotr** [comment]
Just checked, and I am using v0.9.7 of the data importer

**Comment by JC5** [comment]
Thanks. That's the latest version. Your logs, unfortunately, only show the Apache hits. I hope to be able to fix this without needing more details from you. Thanks for the extensive details!

**Comment by Samdotr** [comment]
No problem! I'd be happy to re-run it again with debug logging on the app container if you need any more info. Thanks so much for your help so far 🙂

**Comment by Samdotr** [comment]
Thanks so much for the quick patch! I re-ran using that, and it does indeed show the reason

`Line #0: transactions.0.date: The transactions.0.date field is required. (original value: "")`

This then led me to try importing the first column as Date, rather than Transaction Payment Date, not sure why I didn't try that one before, I feel rather silly now 😄 

After doing that, it does indeed import properly:

`Line #0: Created withdrawal #4 "AppleGreen" (GBP 30.03)`

I'll keep using this development version while trying some bigger imports, once done can I revert to the latest release or should I wait for 0.9.8 to fully release?

Thanks again for your help 🙂

**Comment by JC5** [comment]
I'm fairly confident this will fix things so this will probably be the final version of "0.9.8" before the end of the day (in about 6 hrs).

Thanks for the confirmation!

**Comment by JC5** [comment]
The new release is out, let me know if it works!

**Comment by MurgoloMichele** [comment]
I upgraded to 0.9.8, still getting error on transactions.0.date. My first column is a date in a d/m/Y format with a semicolon as a delimiter that I configured on the configuration page.

Truncated from this point <----|rt-v0.9.8","tags":[],"source_iban":null,"source_number":null,"source_bic":null,"destination_iban":null,"destination_number":null,"destination_bic":null}]} 
[2022-04-12 21:55:06] local.ERROR: Submission error: 0 ["Il campo transactions.0.date è obbligatorio."] 
[2022-04-12 21:55:06] local.DEBUG: This is not a duplicate transaction error  
[2022-04-12 21:55:06] local.DEBUG: Add error on index #218 (line no. 219): transactions.0.date: Il campo transactions.0.date è obbligatorio. (original value: "")  
[2022-04-12 21:55:06] local.DEBUG: Now in App\Services\Shared\Import\Status\SubmissionStatusManager::storeSubmissionStatus(conv-OuRmJ93ST78s): submission_running  
[2022-04-12 21:55:06] local.ERROR: transactions.0.date: Il campo transactions.0.date è obbligatorio. (original value: "")  
[2022-04-12 21:55:06] local.DEBUG: Group is empty, may not have been stored correctly.
